mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -205,6 +210,8 @@ static void put(Configuration config, Table table) {
     config.set(InputFormatConfig.TABLE_LOCATION, table.location());
     config.set(InputFormatConfig.TABLE_SCHEMA, SchemaParser.toJson(table.schema()));
     config.set(InputFormatConfig.PARTITION_SPEC, PartitionSpecParser.toJson(table.spec()));
+    config.set(InputFormatConfig.METADATA_LOCATION,
+        ((HasTableOperations) table).operations().current().metadataFileLocation());
```

Comment:
Same as above. Wondering if the `Table` instance that's passed in can be something other than `BaseTable`, such as `PartitionsTable`, `HistoryTable`, etc.

Reply:
Removed StaticTable and used BaseTable instead. Also checking for BaseTable instead of HasTableOperations